lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -96,7 +96,7 @@ DartType getExpectedType(PostfixExpression node) {
   }
   // as right member of binary operator
   if (parent is BinaryExpression && parent.rightOperand == realNode) {
-    return parent.staticElement.parameters.first.type;
+    return parent.staticElement?.parameters?.first?.type;
```

Comment:
It looks like a big hammer to me, especially here, but also the change above.
Exactly one thing can be `null` here - the `staticElement`.
Using cascading `?.` hides this fact.
Maybe we should ask the element, and branch on if it is `null`.

Reply:
The one above is clumsy to re-write but this  one for sure.

Reply:
Actually.  I can fix.